pkg/yurtmanager/controller/loadbalancerset/viploadbalancer/viploadbalancer_controller.go

```diff
@@ -0,0 +1,290 @@
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to PoolService
+	err = c.Watch(&source.Kind{Type: &netv1alpha1.PoolService{}}, &handler.EnqueueRequestForObject{}, NewPoolServicePredicated())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// +kubebuilder:rbac:groups=network.openyurt.io,resources=poolservices,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reads that state of the cluster for a PoolService object and makes changes based on the state read
+func (r *ReconcileVipLoadBalancer) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	klog.Infof(Format("Reconcile VipLoadBalancer %s/%s", request.Namespace, request.Name))
+
+	// Fetch the PoolService instance
+	poolService := &netv1alpha1.PoolService{}
+	err := r.Get(context.TODO(), request.NamespacedName, poolService)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	copyPoolService := poolService.DeepCopy()
+
+	if poolService.DeletionTimestamp != nil {
+		return r.reconcileDelete(ctx, copyPoolService)
+	}
+
+	if err := r.syncPoolServices(ctx, copyPoolService); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileVipLoadBalancer) syncPoolServices(ctx context.Context, poolService *netv1alpha1.PoolService) error {
+	klog.V(4).Infof(Format("SyncPoolServices VipLoadBalancer %s/%s", poolService.Namespace, poolService.Name))
+	// if not exist, create a new VRID
+	if err := r.handleVRID(poolService); err != nil {
+		return err
+	}
+
+	// Update the PoolService
+	if err := r.Update(ctx, poolService); err != nil {
+		klog.Errorf(Format("Failed to create PoolService %s/%s: %v", poolService.Namespace, poolService.Name, err))
+		return err
+	}
+
+	// sync VRID from the poolservice
+	if err := r.syncVRID(ctx, poolService); err != nil {
+		klog.Errorf(Format("Failed to sync VRID on Pool Service %s/%s: %v", poolService.Namespace, poolService.Name, err))
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileVipLoadBalancer) syncVRID(ctx context.Context, poolService *netv1alpha1.PoolService) error {
+	currentVRIDs, err := r.getCurrentVRID(ctx, poolService)
+	if err != nil {
+		return fmt.Errorf("failed to get current PoolServices: %v", err)
+	}
+
+	r.VRIDManager.SyncVRID(poolService.Labels[network.LabelNodePoolName], currentVRIDs)
+	return nil
+}
+
+func (r *ReconcileVipLoadBalancer) getCurrentVRID(ctx context.Context, poolService *netv1alpha1.PoolService) ([]int, error) {
+	// Get the poolservice list
+	listSelector := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			network.LabelNodePoolName: poolService.Labels[network.LabelNodePoolName],
+		}),
+		FieldSelector: fields.SelectorFromSet(map[string]string{
+			"spec.loadBalancerClass": VipLoadBalancerClass,
```

Comment:
It seems that poolService lister can not support this `FieldSelector`, so how about remove it.